src/main/java/net/snowflake/client/core/SessionUtil.java

```diff
@@ -443,8 +443,18 @@ private static SFLoginOutput newSession(
         }
       } else if (authenticatorType == ClientAuthnDTO.AuthenticatorType.OKTA) {
         data.put(ClientAuthnParameter.RAW_SAML_RESPONSE.name(), tokenOrSamlResponse);
-      } else if (authenticatorType == ClientAuthnDTO.AuthenticatorType.OAUTH
-          || authenticatorType == ClientAuthnDTO.AuthenticatorType.SNOWFLAKE_JWT) {
+      } else if (authenticatorType == ClientAuthnDTO.AuthenticatorType.OAUTH) {
+        data.put(ClientAuthnParameter.AUTHENTICATOR.name(), authenticatorType.name());
+
+        // Fix for HikariCP refresh token issue:SNOW-533673.
+        // If token value is not set but password field is set then
+        // the driver treats password as token.
+        if (loginInput.getToken() != null || loginInput.getPassword() == null)
+          data.put(ClientAuthnParameter.TOKEN.name(), loginInput.getToken());
```

Comment:
Can we throw an  error for a situation where the token is null and the password is also not set. I am not sure if we have null checks for token further down in the flow. It will just be prudent to put a check here.